### ISSUE TITLE
Some fonts render with the wrong font face

### ISSUE THREAD

**Issue by schester44** [tkrkt/text2png#39]
Its hit or miss but it seems like fonts that are more "stylish" or complex are the ones that don't render correctly. A group of fonts within the same family will all render using, what appears to be, the exact same font.

This is occurring on OSX and Freebsd.

Example fonts:
https://drive.google.com/drive/folders/1tBD2RnuEpLHofEdS4Eeg7VvSgk6nt7S1?usp=sharing

The code i'm using:

```js
import text2png from 'text2png'
import fontkit from 'fontkit'

const textToImage = ({ text, file }) => {
	const name = text || fontkit.openSync(file).fullName

	return text2png(name, {
		padding: 10,
		font: `20px ${name}`,
		localFontPath: file,
		localFontName: name
	})
}

export default textToImage
```

I'll continue my research upstream but wanted to drop an issue on this project in case someone has some info related to this.

**Comment by schester44** [comment]
an example of the sort of output i'm getting with those fonts above

https://imgur.com/QDzvhfp

**Comment by schester44** [comment]
It's like the first font is cached and any subsequent calls for similar fonts within the same family are using the first font instead of their own file.

```
localFontPath: file,
localFontName: name
```

the values for these options are different for each file so I would think there would be no relation between the files.

**Comment by schester44** [comment]
I believe I found the issue. Opening these fonts in FontForge, their Preferred Family Name is 'SignPainter' and their Style values are 'Regular'. I believe node-canvas is caching these fonts based on their preferred family name and style. The fonts started rendering correctly once I changed the preferred family name to a unique value.

This seems to be related to this comment here..
https://github.com/Automattic/node-canvas/blob/d2d632402b7032fb0ff755be138d20b87766aff6/examples/font.js#L14

and possibly this section of code here ...
https://github.com/Automattic/node-canvas/blob/master/src/register_font.cc#L130


just need to figure out a way to make it work without using FontForge to modify the values....

**Comment by angelogulina** [comment]
@schester44 I had a similar case and I can confirm your suggestion worked for me.
Therefore I think this can be closed (and thanks so much for providing such insight!)

**Comment by schester44** [comment]
yeah apparently this isn't currently fixable at the node-canvas level (see the referenced issue). Modifying fonts via FontForge isn't really an option for me... open to ideas/suggestions for a fix if anyone comes across this.
